lib/tether/tether.go

```diff
@@ -521,44 +526,50 @@ func (t *tether) Start() error {
 
 		t.setLogLevel()
 
-		if err := t.setHostname(); err != nil {
-			log.Error(err)
-			return err
-		}
+		// TODO: this ensures that we run vm related setup code once
+		// This is temporary as none of those functions are idempotent at this point
+		// https://github.com/vmware/vic/issues/5833
+		initialize.Do(func() {
+			if err = t.setHostname(); err != nil {
+				return
+			}
 
-		// process the networks then publish any dynamic data
-		if err := t.setNetworks(); err != nil {
-			log.Error(err)
-			return err
-		}
-		extraconfig.Encode(t.sink, t.config)
+			// process the networks then publish any dynamic data
+			if err = t.setNetworks(); err != nil {
```

Comment:
Including this here may cause problems later when we (if ever) enable hot add for NICs (to support `docker network connect` on running containers), but for now I don't see any other way to solve the problem at hand.